tests/Benchmarks/System.Text.Formatting/StringFormatterPerf.cs

```diff
@@ -0,0 +1,241 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.Buffers.Text;
+using System.Text;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using BenchmarkDotNet.Attributes;
+using System.Text.Formatting;
+
+namespace System.Text.Formatting.Benchmarks
+{
+    public class StringFormatterPerf
+    {
+        [Params(10, 1000)]
+        public int NumbersToWrite { get; set; }
+        private static ArrayPool<byte> pool = ArrayPool<byte>.Shared;
+
+        [Benchmark]
+        public void InvariantFormatIntDec()
+        {
+            var sb = new StringFormatter(NumbersToWrite, pool);
+            for (int i = 0; i < NumbersToWrite; i++)
+            {
+                sb.Append(((int)(i % 10)));
+            }
+
+            var text = sb.ToString();
```

Comment:
to make sure that the benchmarked code does not get dead code eliminated we always have to return the result from a benchmark. Then BDN can consume it.